### ISSUE TITLE
Questions about“Attention-Based Query Expansion Learning”

### ISSUE THREAD

**Issue by MCC-WH** [filipradenovic/cnnimageretrieval-pytorch#72]
Hi Filip,
I'm very interested in your ECCV paper “Attention-Based Query Expansion Learning” and I tried to implement it by myself. However, I couldn't achieve the good results given in your paper. 

For example, for the roxford5k dataset, all my re-ranking results are much lower than the initial search results. I do use learnable position coding, but it doesn't work. I am not sure if I understand your approach correctly. I am confused whether you are using Auxiliary classification loss with the cluster-ID as the label, which means that images with the same cluster-ID as query are its positive samples.

Besides the above issues, I found that even the network with random initialization achieves higher performance on rparis6k than that reported in the paper. For example, it can reach 88.2 on Medium protocol and 74.4 on hard protocol, which is higher than the results with DBA in the paper. As the training proceeds, the performance on roxford5k is consistently lower than the initial search results, and the results of rparis6k remain almost constant.

**Comment by sakura2233565548** [comment]
I also meet this same issue as @MCC-WH when reproducing your result @filipradenovic ! Could you give me some suggestions~

**Comment by maoyunyao** [comment]
Same issues for me, if possible could you release the source code @filipradenovic
